### ISSUE TITLE
ACMEv2 client includes resource field in JWS payload

### ISSUE THREAD

**Issue by bmw** [certbot/certbot#5708]
Relevant log is:
```
JWS payload:
{
  "termsOfServiceAgreed": true, 
  "resource": "new-reg"
}
Sending POST request to https://acme-staging-v02.api.letsencrypt.org/acme/new-acct:
{
  "protected": "eyJub25jZSI6ICJUMDN3d195UjcxV2haN2ZsbGFHVlVQUF8tejkxcVhKQ1FFaFFwQWp6eTNNIiwgInVybCI6ICJodHRwczovL2FjbWUtc3RhZ2luZy12MDIuYXBpLmxldHNlbmNyeXB0Lm9yZy9hY21lL25ldy1hY2N0IiwgImp3ayI6IHsiZSI6ICJBUUFCIiwgImt0eSI6ICJSU0EiLCAibiI6ICI2QnlTWHVwbnpHU1Y0ME53UHhqRTdSdEp3cFlvNnJyUWNEX0xOMm94RFIzTFQ0N0JjZ1AySHhXR2xSTDV1YWtjWHJWaHlScjd4aUJkNDUzSnBvU2VxR1RnS0tKVVp2eDVnd09tNzg4SE9pSEJDdklva0ljNzBjN3JQTmFrR0tZdXp6SVlUTmR4b2w3dWdSaVhxdUZOQnRZeXlLeDRnVUx6Y1QzYmh4dVdfeldISC1HSGVoNXA4Qjg1TE9KZEx5WlV2djNjQWxYLUYtYmJJX3VqRm5rc0ZhVTI3Z2s0RFlldllpRjN0SEtqZXQ4UXlWeTYwRk1wRFdkaUZhZTJVeklFNXZiczRPX1FvVTBZVWZfbTVMbHFQVmF0TWk0VDVFdFVOZkJGYWp2U0FBcTdoVkg4bmJhbzdVMVVQVVJsQVFZc2otSVFaNlFoblBVWExRZjMtVGozTXcifSwgImFsZyI6ICJSUzI1NiJ9", 
  "payload": "ewogICJ0ZXJtc09mU2VydmljZUFncmVlZCI6IHRydWUsIAogICJyZXNvdXJjZSI6ICJuZXctcmVnIgp9", 
  "signature": "k1XUZQUo0PBPvsVBr12VdsUhqZRwRXrjGnUkxCR2NSW_5aTqPCySwcrkeiPWAkHF6-iy020vkSZXnZuMYEfqyx_NrCeSmvwRQmMG1qeedB2t0SD1W4kpeNDxgSdHvX4ks57d-LFXoleuYQ8hp5cK8Cvnhn5vDPasDgiyoEErnphS5AK4Vso2FUju7Yw3B2NTfhogobljdeOUiaYnovE2kHNLhztyD1HyA4VPAhyw_excbbOxmt2F35Dr6ySRrBUkK_iHvmT7kDjZBNzLdcQS8rHqqvfqADHni-zsjADgee4o3k04df66MK4sb_hdIx1ikcvAt187q2tahn-FzO09Ew"
}
```
This was required in older versions of the spec but isn't referenced in the current spec so we should probably stop sending the value when using ACMEv2.

@jsha, am I correct that this is cruft that can be removed?

**Comment by cpu** [comment]
> @jsha, am I correct that this is cruft that can be removed?

Yes: the "resource" field isn't required in ACME v2 because the full request URL is in the JWS protected headers.

_(Stealing @jsha's thunder because I saw this notification first :laughing:)_

**Comment by bmw** [comment]
@cpu, I regularly find it impressive how on top of your GH notifications you are. Thanks for the sanity check!

**Comment by bmw** [comment]
This wasn't fixed in #6271.

**Comment by cpu** [comment]
A fix for this issue can be tested with `pebble -strict` as of [Pebble v2.1.0](https://github.com/letsencrypt/pebble/releases/tag/v2.1.0).

**Comment by bmw** [comment]
This was fixed in #7176.
